### ISSUE TITLE
Analysis server getHover doesn't work for locations inside SDK files

### ISSUE THREAD

**Issue by DanTup** [dart-lang/sdk#27241]
This might not be a bug, but I think it'd be good to support. If I hover over `ArgumentError` in my own code, I get a tooltip. If I Go to Definition on the class (which opens `lib\core\errors.dart`) and then try hovering over things inside the SDK file, no hover information is returned:

```text
[16:31:06]: ==> {"id":"135","method":"analysis.getHover","params":{"file":"m:\\Apps\\Dart\\lib\\core\\errors.dart","offset":4171}}
[16:31:06]: <== {"id":"135","result":{"hovers":[]}}
```

Since the SDK is shipped in source form and the user often ends up in it (via Go to Definition, or stepping in with the debugger) it would be nice if this worked.

@scheglov I don't know if this might be related to #27214 which was a similar failure for `getNavigation` (or if the same fix might even fix this?)

**Comment by scheglov** [comment]
Yes, `getNavigation` and other semantic features are all based on fully resolved units.
So, the fix for #27214 fixes also this issue.
It works for me now.
So, I'm closing this issue.

**Comment by DanTup** [comment]
@scheglov Excellent, thanks for confirming!
